### DISCUSSION TITLE
Speed calculation

### DISCUSSION THREAD

**Discussion by michvhf** [shark92651/LaserParamsConverter#14] [Q&A]
Not being able to run this program I'm at a loss when reading the code wrt the speed calculation.   maxPower is a variable and I assume it's for the max percent you want the laser to put out.  If outPower > maxPower then calculate the speed and make outPower the same as maxPower.  But outSpeed never changes wrt maxPower.   It's the same if maxPower is 100% or 50%.   Am I missing something here?

**Comment by shark92651** [comment]
MaxPower is primarily for use with CO2 lasers where running at 100% can affect the life of the tube. For fiber lasers, 100% is fine. When the calculated power is above MaxPower, the power is set to MaxPower and the speed is reduced. The power will never go above MaxPower.

**Reply by michvhf** [reply]
Source:  60 watt laser, 63% power, 200 mm/s
Dest: 38 watt laser
Lens doesn't matter, as long as they're the same.
In this scenario, The power will be 100% (99.4% actually) and the speed 200 mm/s.
Now change Dest to 37 watt laser.
In this scenario, the power will be 100% (102.16% before being dropped to 100%) and the speed will be 123.33 mm/s.   Quite a drop in speed for 1 watt.
